packages/hooks-react/src/useBillingPortal.tsx

```diff
@@ -0,0 +1,31 @@
+import { useMutation } from 'react-query';
+import { useCallback, useMemo, useState } from 'react';
+import { getModule } from '@jwp/ott-common/src/modules/container';
+import CheckoutController from '@jwp/ott-common/src/controllers/CheckoutController';
+
+const useBillingPortal = () => {
+  const checkoutController = getModule(CheckoutController);
+
+  const [isRedirecting, setRedirecting] = useState(false);
+
+  const billingPortalUrlRequest = useMutation({
+    mutationKey: 'billing-portal',
+    mutationFn: () => checkoutController.generateBillingPortalUrl(window.location.href),
+  });
+
+  const redirectToBillingPortal = useCallback(async () => {
+    const billingPortalUrl = await billingPortalUrlRequest.mutateAsync();
+
+    setRedirecting(true);
+
+    if (billingPortalUrl) {
+      window.location.href = billingPortalUrl;
+    }
+  }, [billingPortalUrlRequest]);
```

Comment:
This assumes that this hook is only used in a browser. The [hooks-react](https://github.com/jwplayer/ott-web-app/blob/develop/docs/workspaces.md#react-hooks) is platform-agnostic and can't use `window.location` directly.

Reply:
Fixed this.